_posts/2021-12-28-trino-2021-a-year-of-growth.md

```diff
@@ -0,0 +1,347 @@
+* Changed minimum number of speculative executions from 2 to 1. 
+
+### Clickhouse connector
+
+* Added [ClickHouse connector]({{site.url}}/docs/current/connector/clickhouse.html).
+* Improved performance of aggregation queries by computing aggregations within 
+ClickHouse. Currently, the following aggregate functions are eligible for
+pushdown: `count`, `min`, `max`, `sum` and `avg`.
+* Added support for dropping columns.
+* Map ClickHouse `UUID` columns as `UUID` type in Trino instead of `VARCHAR`.
+
+### HDFS, S3, Azure and cloud object storage systems
+
+A core use case of Trino uses the Hive and Iceberg connectors to connect to
+a data lake. These connectors differ from most as Trino is the sole query engine
+as opposed to the client calling another system. Here are some changes that
+for these connectors:
+
+* Fixed queries on empty Iceberg tables without a snapshot ID that were created by Spark.
+* Updated to Iceberg 0.11.0 behavior for transforms of dates and timestamps before 1970.
+* Added procedure `system.flush_metadata_cache()` to flush metadata caches.
+* Avoid generating splits for empty files.
+* Sped up Iceberg query performance when dynamic filtering can be leveraged.
+* Increased Iceberg performance when reading timestamps from Parquet files.
+* Improved Iceberg performance for queries on nested data through dereference pushdown.
+* Added support for `INSERT OVERWRITE` operations on S3-backed tables.
+* Made the Iceberg `uuid` type available.
+* Trino views made available in Iceberg.
+
+
+### Elasticsearch connector
+
+* Added support for reading fields as `json` values.
+* Fixed failure when documents contain fields of unsupported types.
+* Added support for `scaled_float` type.
+* Added support for assuming an IAM role.
+* Fixed potential incorrect results when queries contain an `IS NULL` predicate.
+* Added retry requests with backoff when Elasticsearch is overloaded.
+
+### Kafka connector
+
+* Added support for TLS security protocol.
+* Fixed querying of schema registry tables with references in their schema.
+* Fixed listing of schema registry tables having ambiguous subject name in lower case.
+
+### MemSQL connector
+
+* Allowed limiting the size of the metadata cache.
+* Added partial support for `DELETE` statement where predicate can be fully pushed 
+down to the remote datasource.
+* Added support for MemSQL 3.2.
+
+### MongoDB connector
+
+* Added [`timestamp_objectid()`]({{site.url}}/docs/current/connector/mongodb.html#timestamp_objectid)
+function.
+* Enabled `mongodb.socket-keep-alive` config property by default.
+* Add support for `json` type.
+* Support reading MongoDB `DBRef` type.
+* Allow skipping creation of an index for the `_schema` collection, if it 
+already exists.
+* Added support to redact the value of `mongodb.credentials` in the server log.
+* Added support for dropping columns.
+
+### MySQL connector
+
+* Added support for reading and writing `timestamp` values with precision higher
+than three.
```

Comment:
```suggestion
* Added support for reading and writing `timestamp` values with precision higher
than 3.
```

Reply:
@mosabua suggested the opposite. I think in general when we're not writing documentation, we spell out letters.